core/headerchain.go

```diff
@@ -145,10 +145,14 @@ func (hc *HeaderChain) WriteHeader(header *types.Header) (status WriteStatus, er
 	var (
 		hash     = header.Hash()
 		number   = header.Number.Uint64()
-		localTd  *big.Int
 		externTd *big.Int
 	)
 
+	if cannonicalHash := hc.GetCanonicalHash(number); (cannonicalHash != common.Hash{} && cannonicalHash != hash) {
```

Comment:
Can you adjust the function comment? It talks a lot about re-orgs and side chains which ofc is no longer relevant